test/common/router/router_test.cc

```diff
@@ -90,6 +90,9 @@ class RouterTestBase : public testing::Test {
     router_.downstream_connection_.local_address_ = host_address_;
     router_.downstream_connection_.remote_address_ =
         Network::Utility::parseInternetAddressAndPort("1.2.3.4:80");
+
+    // make the "system time" non-zero, because 0 is considered invalid by DateUtil
+    test_time_.sleep(std::chrono::milliseconds(50));
```

Comment:
You could use `setSystemTime` or `setMonotonicTime` methods to explicitly set the "start time", which might read a bit clearer about what you're trying to do here.

Reply:
agreed, made the change